plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTypeMapping.java

```diff
@@ -0,0 +1,597 @@
+    }
+
+    @DataProvider
+    public Object[][] sessionZonesDataProvider()
+    {
+        return new Object[][] {
+                {UTC},
+                {jvmZone},
+                // using two non-JVM zones so that we don't need to worry what Cassandra system zone is
+                {vilnius},
+                {kathmandu},
+                {ZoneId.of(TestingSession.DEFAULT_TIME_ZONE_KEY.getId())},
+        };
+    }
+
+    @Test
+    public void testCassandraTimeUuid()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("timeuuid", "NULL", UUID, "CAST(NULL AS UUID)")
+                .addRoundTrip("timeuuid", "50554d6e-29bb-11e5-b345-feff819cdc9f", UUID, "CAST('50554d6e-29bb-11e5-b345-feff819cdc9f' AS UUID)")
+                .execute(getQueryRunner(), cassandraCreateAndInsert("tpch.test_timeuuid"));
+    }
+
+    @Test
+    public void testUuid()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("uuid", "NULL", UUID, "CAST(NULL AS UUID)")
+                .addRoundTrip("uuid", "114514ea-0601-1981-1142-e9b55b0abd6d", UUID, "CAST('114514ea-0601-1981-1142-e9b55b0abd6d' AS UUID)")
+                .execute(getQueryRunner(), cassandraCreateAndInsert("tpch.test_uuid"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("uuid", "NULL", UUID, "CAST(NULL AS UUID)")
+                .addRoundTrip("uuid", "UUID '114514ea-0601-1981-1142-e9b55b0abd6d'", UUID, "CAST('114514ea-0601-1981-1142-e9b55b0abd6d' AS UUID)")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_uuid"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_uuid"));
+    }
+
+    @Test
+    public void testCassandraTuple()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("tuple<int, text>", "NULL", anonymousRow(INTEGER, VARCHAR), "CAST(NULL AS ROW(INTEGER, VARCHAR))")
+                .addRoundTrip("tuple<int, text>", "(3, 'hours')", anonymousRow(INTEGER, VARCHAR), "CAST(ROW(3, 'hours') AS ROW(INTEGER, VARCHAR))")
+                .addRoundTrip("tuple<list<text>>", "(['Cassandra'])", anonymousRow(VARCHAR), "CAST(ROW('[\"Cassandra\"]') AS ROW(VARCHAR))")
+                .addRoundTrip("tuple<map<text, text>>", "({'connector':'Cassandra'})", anonymousRow(VARCHAR), "CAST(ROW('{\"connector\":\"Cassandra\"}') AS ROW(VARCHAR))")
+                .addRoundTrip("tuple<set<text>>", "({'Cassandra'})", anonymousRow(VARCHAR), "CAST(ROW('[\"Cassandra\"]') AS ROW(VARCHAR))")
+                .addRoundTrip("tuple<tuple<int, text>>", "((3, 'hours'))", anonymousRow(anonymousRow(INTEGER, VARCHAR)), "CAST(ROW(ROW(3, 'hours')) AS ROW(ROW(INTEGER, VARCHAR)))")
+                .execute(getQueryRunner(), cassandraCreateAndInsert("tpch.test_tuple"));
+    }
+
+    @Test
+    public void testCassandraUdt()
+    {
+        session.execute("CREATE TYPE tpch.phone (country_code int, number text)");
```

Comment:
Let's add `session.execute("DROP TYPE IF EXISTS tpch.phone");` before creating the type. We may want make `CassandraServer` reusable in the future.